front/src/applications/editor/tools/switchEdition/types.ts

```diff
@@ -77,15 +80,46 @@ const trackNodeTypeOrder = [
 ];
 
 export const useSwitchTypes = (infraID: number | undefined) => {
-  const { data } = osrdEditoastApi.endpoints.getInfraByInfraIdSwitchTypes.useQuery(
-    { infraId: infraID as number },
-    { skip: !infraID }
+  const [data, setData] = useState<SwitchType[]>([]);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [error, setError] = useState<Error | null>(null);
+  const [getInfraSwitchTypes] = osrdEditoastApi.endpoints.getInfraByInfraIdSwitchTypes.useLazyQuery(
+    {}
```

Comment:
I think you can use rtk query built-in states to avoid creating new state for loading an error ?

```suggestion
  const [getInfraSwitchTypes, { isLoading, isError }] =
    osrdEditoastApi.endpoints.getInfraByInfraIdSwitchTypes.useLazyQuery({});
```